### DISCUSSION TITLE
Confusing validation message

### DISCUSSION THREAD

**Discussion by vassilvk** [cue-lang/cue#2836] [General]
Consider the following:

`schema.cue`:
```
package my_package

// Top enum for all possible platforms.
#Platform: "linux/amd64" | "linux/arm64" | "windows/amd64" | "windows/arm64"

// Dynamic enum evaluated based on the actual set of platforms the container image was built for.
#ContainerImagePlatform : or(deployment_meta.container_image_platforms)

// Information injected by an external tool.
#DeploymentMeta: {
  // Container image platform - expected to be at least one of all the possible platforms.
  container_image_platforms: [#Platform, ...#Platform]
}

// User specified configuration.
#Workload: {
  deploy_platforms: [#ContainerImagePlatform, ...#ContainerImagePlatform]
}

// Concrete configuration sections.
deployment_meta: #DeploymentMeta
workload: #Workload
```

`values.yaml`:
```
# Generated by external tooling.
deployment_meta:
  container_image_platforms:
    - linux/amd64
    - linux/arm64

# Provided by user.
workload:
  deploy_platforms:
    - windows/amd64
```

Here's the goal:

- `deployment_meta.container_image_platforms` should be one of `["linux/arm64", "linux/amd64", "windows/amd64", "windows/arm64"]`.
- `workload.deployment_platforms` should match one of the values in `deployment_meta.container_image_platforms`.

The above YAML file contains an intentional validation issue to demonstrate the problem.
List `workload.deployment_platforms` contains `windows/amd64` which is not on the list of items in `deployment_meta.container_image_platforms`.

Running `vet values.yaml schema.cue` generates the following:
```
workload.deploy_platforms.0: 3 errors in empty disjunction:
workload.deploy_platforms.0: conflicting values "linux/amd64" and "windows/amd64":
    ./schema.cue:17:22
    ./schema.cue:22:11
    ./values.yaml:4:7
workload.deploy_platforms.0: conflicting values "linux/arm64" and "windows/amd64":
    ./schema.cue:4:28
    ./schema.cue:12:31
    ./schema.cue:17:22
    ./schema.cue:22:11
workload.deploy_platforms.0: conflicting values "windows/arm64" and "windows/amd64":
    ./schema.cue:4:62
    ./schema.cue:12:31
    ./schema.cue:17:22
    ./schema.cue:22:11
```

I'm not sure I understand where the third message (about `windows/arm64`) is coming from, as `windows/arm64` is not in the `deployment_meta.container_image_platforms` list.

**Comment by jpluscplusm** [comment]
Hey @vassilvk 👋 

So I was confused by this as well, and I went digging :-) I've raised https://github.com/cue-lang/cue/issues/2912 to flag the behaviour as a bug. 

I've phrased the issue in terms of the (seemingly) superfluous and confusing conflicting value flagged by the disjunction, so hopefully the [new evaluator](https://github.com/orgs/cue-lang/projects/18) that's being developed can prefer the form of the error message that would be useful here.

However, given that it's *also* an ordering-related issue, I'm not sure what fix might ultimately be proposed, and if it would make the output less confusing. I hope it will, of course!

I'm not sure if you can affect the ordering sufficiently, in your more complete example, to avoid the confusing, conflicting value. I wasn't able to make that work, here, but maybe you'll have better luck since you control the wider shape of your CUE and your data.
